snuba/query/snql/parser.py

```diff
@@ -1317,4 +1395,26 @@ def parse_snql_query(
 
     # Validating
     _post_process(query, VALIDATORS)
+
+    if isinstance(query, LogicalQuery):  # avoid composite for now
```

Comment:
Why do we sort the logical query at parsing time instead of the clickhouse query? We decide whether to use the cache and compute the caching key based on the clickhouse query. Right now I cannot think about a processor that would re-sort the expression in a non consistent way but it may happen. 
We do re-sort the condition in many places but they should all yield the same result if the input is the same. ... hopefully.

Reply:
The reason I did it here is related to your comment above, where you point out a place we add an array/tuple argument to the query. If I put the sorting here, everything that happens after is (probably) deterministic and I there are fewer chances of messing up the query.

Reply:
If you sorted the query right before using the cache instead you would be able to have a valid success metric as you could get the cache key of the original query and the cache key of the new one, then you could try to find both in cache. That seems the only way to validate the results of this experiments, as relying on the cache hit rate will likely be close to useless (it changes a lot during the day because of usage pattern fluctuations).